geomstats/geometry/hyperbolic.py

```diff
@@ -670,28 +676,32 @@ def log(self, point, base_point):
 
             add_base_point = self.mobius_add(-base_point, point)
 
-            norm_add = gs.to_ndarray(gs.norm(add_base_point, -1), 2, -1)
+            norm_add = gs.to_ndarray(gs.linalg.norm(
+                add_base_point, axis=-1), 2, -1)
             norm_add = gs.repeat(norm_add, base_point.shape[-1], -1)
-            norm2_base_point = gs.to_ndarray(gs.sum(base_point**2, -1), 2, -1)
+            norm2_base_point = gs.to_ndarray(gs.linalg.norm(
+                base_point, axis=-1), 2, -1)
             norm2_base_point = gs.repeat(norm2_base_point,
                                          base_point.shape[-1], -1)
 
-            log = (1 - norm2_base_point) * gs.arctanh(norm_add)\
+            log = (1 - norm2_base_point**2) * gs.arctanh(norm_add)\
                 * (add_base_point / norm_add)
 
             mask_0 = gs.all(gs.isclose(norm_add, 0))
             log[mask_0] = 0
+            log[gs.isnan(log)] = 0
 
             return log
         else:
             raise NotImplementedError(
                     'log is only implemented for ball and extrinsic')
 
     def mobius_add(self, point_a, point_b):
-        """Compute the mobius addition of two points.
+        """Mobius addition of two points.
```

Comment:
Could you also add a short formula describing the mobius addition?

Reply:
Added formula using .. math:: operator. Not sure how to test if that would work well.